experiments/rsc/src/worker.tsx

```diff
@@ -136,32 +156,51 @@ export default {
         return Response.redirect(referer, 303);
       }
 
-      const pathname = new URL(request.url).pathname as keyof typeof routes;
-      const Page = routes[pathname];
-      if (!Page) {
-        // todo(peterp, 2024-11-25): Return not found page, if exists
-        return new Response("Not found", { status: 404 });
-      }
+      const renderPage = async (Page: any, props = {}) => {
+        const rscPayloadStream = renderToRscStream(<Page {...props} />);
 
-      const rscPayloadStream = renderToRscStream(<Page />);
+        if (isRSCRequest) {
+          return new Response(rscPayloadStream, {
+            headers: { "content-type": "text/x-component; charset=utf-8" },
+          });
+        }
+        const [rscPayloadStream1, rscPayloadStream2] = rscPayloadStream.tee();
 
-      if (isRSCRequest) {
-        return new Response(rscPayloadStream, {
-          headers: { "content-type": "text/x-component; charset=utf-8" },
+        const htmlStream = await transformRscToHtmlStream({
+          stream: rscPayloadStream1,
+          Parent: App,
         });
-      }
 
-      const [rscPayloadStream1, rscPayloadStream2] = rscPayloadStream.tee();
+        const html = htmlStream.pipeThrough(
+          injectRSCPayload(rscPayloadStream2),
+        );
+        return new Response(html, {
+          headers: { "content-type": "text/html" },
+        });
+      };
 
-      const htmlStream = await transformRscToHtmlStream({
-        stream: rscPayloadStream1,
-        Parent: App,
-      });
+      const pathname = new URL(request.url).pathname as keyof typeof routes;
+      const Page = routes[pathname];
+      if (!Page) {
+        // Check if it matches the tradesmen dynamic route pattern
+        const tradesmenMatch = pathname.match(/^\/tradesmen\/(.+)$/);
+        const tradesmanMatch = pathname.match(/^\/tradesman\/(.+)$/);
+        if (tradesmenMatch) {
+          const profession = tradesmenMatch[1];
+          return renderPage(routes["/tradesmen/:profession"], {
+            params: { profession },
+          });
+        }
+        if (tradesmanMatch) {
+          const id = tradesmanMatch[1];
+          return renderPage(routes["/tradesman/:id"], {
```

Comment:
Nice! @peterp maybe there's an opportunity here now for us to figure out how routing should work and/or how the API should look.